### PR TITLE
Update to version 0.9.0 and add versioning tests

### PR DESCRIPTION
**Pull request by asiniscalchi**
## Type
enhancement, tests

___

## Description
This PR includes several enhancements and tests:
- The `spec_version` has been updated from 800 to 900 in the runtime configuration.
- A new test module `spec_version` has been added to test the versioning system.
- The `RUNTIME_SPEC_VERSION` has been updated from 800 to 900 in the e2e tests configuration.
- The version has been updated to 0.9.0 in the main `Cargo.toml` file, and the `workspace` field has been added.
- The version of several packages has been set to use the workspace version.

___

## PR changes walkthrough
<table><thead><tr><th></th><th>Relevant files&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </th></tr></thead><tbody><tr><td><strong>Configuration changes</strong></td><td><details><summary>11 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>lib.rs&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        ownership-chain/runtime/src/lib.rs<br><br>

**The `spec_version` has been updated from 800 to 900.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/258/files#diff-b0f69a71ca20f9dbda0917139a7a931cd6503375c8833428c8ac431ea693aef4"> +1/-1</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>config.ts&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        ownership-chain/e2e-tests/tests/config.ts<br><br>

**The `RUNTIME_SPEC_VERSION` has been updated from 800 to 900.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/258/files#diff-02c263a6e8b3847f277a70f0ea700a1184eb1aa7742e62b25a5329d67edc8a63"> +1/-1</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Cargo.toml&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        Cargo.toml<br><br>

**The version has been updated to 0.9.0. The `workspace` field <br>has been added.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/258/files#diff-2e9d962a08321605940b5a657135052fbcef87b5e360662bb527c96d9a615542"> +1/-0</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Cargo.toml&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        ownership-chain/node/Cargo.toml<br><br>

**The version has been set to use the workspace version.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/258/files#diff-a95acab9d237247f78755bd16f676aebbc0591ffb4200abbadb951bd28d18be6"> +1/-1</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Cargo.toml&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        ownership-chain/pallets/laos-evolution/Cargo.toml<br><br>

**The version has been set to use the workspace version.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/258/files#diff-b21e921ed8ed2ed2bbf6673dfe24a7093b1a4aa973d3515570a874ee3ea6764c"> +1/-1</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Cargo.toml&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        ownership-chain/precompile/evolution-collection-factory/Cargo.toml<br><br>

**The version has been set to use the workspace version.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/258/files#diff-85349d448f7665a8e7282a2387dbc75b27d7a4e14f9727978dbd0378aa243307"> +1/-1</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Cargo.toml&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        ownership-chain/precompile/evolution-collection/Cargo.toml<br><br>

**The version has been set to use the workspace version.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/258/files#diff-30ea770bf8ca64aad723ea8ab492a57e18eae9a9390570f35b9746d6e5138ad8"> +1/-1</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Cargo.toml&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        ownership-chain/precompile/utils/Cargo.toml<br><br>

**The version has been set to use the workspace version.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/258/files#diff-025f8df1213c0e9b863fbbc675434f7eea792cec150a78cdb604fa333a2f1f03"> +1/-1</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Cargo.toml&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        ownership-chain/precompile/utils/macro/Cargo.toml<br><br>

**The version has been set to use the workspace version.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/258/files#diff-dc7bd105ac23656ca662c30e038de65a7383bb5102e08753c422919c9e14b175"> +1/-1</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Cargo.toml&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        ownership-chain/primitives/Cargo.toml<br><br>

**The version has been set to use the workspace version.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/258/files#diff-1254592a14dcc0a9783c990549ceaa96c04ee38332f4e8e8f1c869f8521592c9"> +1/-1</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Cargo.toml&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        ownership-chain/runtime/Cargo.toml<br><br>

**The version has been set to use the workspace version.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/258/files#diff-a6f1e3deb5d45fdcea9465ba45b4c048309ff47223bd15d61c1fe1340e98200f"> +1/-1</a></td>

</tr>                    
</table></details></td></tr><tr><td><strong>Tests</strong></td><td><details><summary>2 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>mod.rs&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        ownership-chain/runtime/src/tests/mod.rs<br><br>

**A new test module `spec_version` has been added.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/258/files#diff-ac443d4befbe7cc699f85f0db740bd8448d68c43ba09e9b7e30f6e68a0740eeb"> +1/-0</a></td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>spec_version.rs&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </strong></summary>
      <ul>
        ownership-chain/runtime/src/tests/spec_version.rs<br><br>

**A new file has been added that contains tests for the <br>versioning system. It includes tests for version format, <br>conversion, and consistency with the package version.**
</ul>
    </details>
  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/258/files#diff-c270cfbdf701526a7448804c3f4e4f6a7f28162238e3b7573eff7d54f3fe8aea"> +90/-0</a></td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

### CODE REVIEW

File: ownership-chain/runtime/src/tests/spec_version.rs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+/// or an `Error` if the input format is incorrect or any part of the version
+/// is 100 or greater.
+///
+/// # Examples
+/// ```
+/// let version = "1.2.3";
+/// let int_version = version_to_int(version).unwrap();
+/// assert_eq!(int_version, 10203);
+/// ```
+fn version_to_int(version: &str) -> Result<u32, Error> {
```
Comment (magecnion) [inline ownership-chain/runtime/src/tests/spec_version.rs:25]: I suggest to find a better name that gives more context of its purpose. (e.g.: `parse_cargo_pkg_version_to_runtime`). Also it is not needed to include the returned value `int` as part of the name.

File: ownership-chain/runtime/src/tests/spec_version.rs [hunk 2]
```diff
@@ -0,0 +1,108 @@
+use crate::VERSION;
```
Comment (magecnion) [inline ownership-chain/runtime/src/tests/spec_version.rs:1]: Why did you implement this inside `tests` folder? I would move this file to `runtime` module instead

**Comment by dastanfv** [comment]
I am also lost why we need any other test than `test_runtime_version_should_be_derived_from_package_version`. All the others are simply testing the utility/test function `version_to_int`, they don't test any runtime config. I would suggest removing that function and all the other tests that test it, and simply have the logic of the function in the test_runtime_version_should_be_derived_from_package_version`. And also rename it to something more generic, f.e `test_runtime_version_sanity`

File: ownership-chain/runtime/src/tests/spec_version.rs (hunk 2 above)
Comment (dastanfv) [reply ownership-chain/runtime/src/tests/spec_version.rs:1]: runtime tests are consolidated in the `tests` folder

Comment (magecnion) [reply ownership-chain/runtime/src/tests/spec_version.rs:1]: it includes logic not only tests, isn't it?

Comment (dastanfv) [reply ownership-chain/runtime/src/tests/spec_version.rs:1]: you mean `version_to_int` function? if yes, see my comment below

File: ownership-chain/runtime/src/tests/spec_version.rs (hunk 1 above)
Comment (asiniscalchi) [reply ownership-chain/runtime/src/tests/spec_version.rs:25]: I don't agree in the `cargo_pkg` part cause it is not what the function does.
maybe `convert_version_to_number` might be better

**Comment by asiniscalchi** [comment]
Closed cause the team is not aligned

**Pull request closed without merging**